detect_secrets/plugins/cloudant.py

```diff
@@ -0,0 +1,131 @@
+    cloudant_api_url = r'cloudant\.com'
+    denylist = [
+        RegexBasedDetector.assign_regex_generator(
+            prefix_regex=cl,
+            secret_keyword_regex=cl_key_or_pass,
+            secret_regex=cl_pw,
+        ),
+        RegexBasedDetector.assign_regex_generator(
+            prefix_regex=cl,
+            secret_keyword_regex=cl_key_or_pass,
+            secret_regex=cl_api_key,
+        ),
+        re.compile(
+            r'{http}{cl_account}{colon}{cl_pw}{at}{cl_account}{dot}{cloudant_api_url}'.format(
+                http=http,
+                colon=colon,
+                cl_account=cl_account,
+                cl_pw=cl_pw,
+                at=at,
+                dot=dot,
+                cloudant_api_url=cloudant_api_url,
+            ),
+            flags=re.IGNORECASE,
+        ),
+        re.compile(
+            r'{http}{cl_account}{colon}{cl_api_key}{at}{cl_account}{dot}{cloudant_api_url}'.format(
+                http=http,
+                colon=colon,
+                cl_account=cl_account,
+                cl_api_key=cl_api_key,
+                at=at,
+                dot=dot,
+                cloudant_api_url=cloudant_api_url,
+            ),
+            flags=re.IGNORECASE,
+        ),
+    ]
+
+    def verify(self, token, content):
+
+        hosts = find_account(content)
+        if not hosts:
+            return VerifiedResult.UNVERIFIED
+
+        for host in hosts:
+            return verify_cloudant_key(host, token)
+
+        return VerifiedResult.VERIFIED_FALSE
+
+
+def find_account(content):
+    opt_hostname_keyword = r'(?:hostname|host|username|id|user|userid|user-id|user-name|' \
+        'name|user_id|user_name|uname|account)'
+    account = r'(\w[\w\-]*)'
+    opt_basic_auth = r'(?:[\w\-:%]*\@)?'
+
+    regexes = (
+        RegexBasedDetector.assign_regex_generator(
+            prefix_regex=CloudantDetector.cl,
+            secret_keyword_regex=opt_hostname_keyword,
+            secret_regex=account,
+        ),
+        re.compile(
+            r'{http}{opt_basic_auth}{cl_account}{dot}{cloudant_api_url}'.format(
+                http=CloudantDetector.http,
+                opt_basic_auth=opt_basic_auth,
+                cl_account=account,
+                cl_api_key=CloudantDetector.cl_api_key,
+                dot=CloudantDetector.dot,
+                cloudant_api_url=CloudantDetector.cloudant_api_url,
+            ),
+            flags=re.IGNORECASE,
+        ),
+    )
+
+    return [
+        match
+        for line in content.splitlines()
+        for regex in regexes
+        for match in regex.findall(line)
+    ]
+
+
+def verify_cloudant_key(hostname, token):
+    try:
+        headers = {'Content-type': 'application/json'}
+        request_url = 'https://{hostname}:' \
+            '{token}' \
+            '@{hostname}.' \
+            'cloudant.com'.format(
+                hostname=hostname,
+                token=token,
+            )
+
+        response = requests.get(
```

Comment:
Can you do the specific exception thing here?

Rest of the PR ✅ ⛵️ 🚢 👍

Reply:
Comment addressed.